### PR TITLE
[tmva] Add RTensor constructor from a numpy array

### PR DESCRIPTION
**Pull request by lmoneta**
This PR allows to convert directly a RTensor from a Numpy array without the need to use the shape.
Example:
```
import ROOT
import numpy as np
 x = np.array([[1.,2.],[3.,4.]])
r = ROOT.TMVA.Experimental.RTensor('double')(x)
print(r.GetShape())
```

### CODE REVIEW

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_rtensor.py [hunk 1]
```diff
@@ -114,6 +113,25 @@ def RTensorGetitem(self, idx):
         idxVec[i] = x
     return self(idxVec)
 
+def RTensorInit(self, *args):
+
+    if (len(args) == 1) :
+        try:
+            import numpy as np
+        except:
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_rtensor.py:121]: I think the error should be more explicit:
```python
except ImportError:
    raise ImportError("Failed to import numpy in RTensor constructor")
```

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_rtensor.py [hunk 2]
```diff
@@ -114,6 +113,25 @@ def RTensorGetitem(self, idx):
...
+        if isinstance(args[0], np.ndarray):
+            data = args[0]
+            #conversion from numpy to buffer float/double * works only if C order
+            if (data.flags.c_contiguous):
+                shape = data.shape
+                from cppyy.gbl import TMVA
+                layout = TMVA.Experimental.MemoryLayout.RowMajor
+                return self._original_init_(data,shape,layout)
+            else:
+                print("Error- cannot convert a Numpy array in Fortran Format to a C array")
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_rtensor.py:132]: With this `if-else` statement, we are just printing an error message but not actually raising an error. I don't think this is the behaviour we want right? If the user is trying to convert a numpy array to RTensor and something goes wrong, the application should stop. Consider changing the `else` block to
```python
else:
    raise ValueError("Can only convert C-contiguous Numpy arrays to RTensor but input array is Fortran-contiguous")
```

**Review by vepadulano — APPROVED** [review]
Looks fine overall, see minor comments

**Comment by lmoneta** [comment]
@phsft-bot build just on windows10/cxx14, ROOT-ubuntu2004/default

**Pull request merged**